src/tests/test_stashcache.py

```diff
@@ -43,6 +67,58 @@ def test_allowedVO_excludes_LIGO_and_ANY_for_ligo_inclusion(self, client: flask.
     def test_None_fdqn_isnt_error(self, client: flask.Flask):
         stashcache.generate_cache_authfile(global_data, None)
 
+    def test_origin_grid_mapfile_nohost(self, client: flask.Flask):
+        text = stashcache.generate_origin_grid_mapfile(global_data, "", suppress_errors=False)
+        for line in text.split("\n"):
+            assert EMPTY_LINE_REGEX.match(line), f'Unexpected text "{line}".\nFull text:\n{text}\n'
+
+    def test_origin_grid_mapfile_with_host(self, client: flask.Flask):
+        text = stashcache.generate_origin_grid_mapfile(global_data, "origin-auth2001.chtc.wisc.edu",
```

Comment:
I see this as more of an integration test than a unit test; as long as it doesn't depend on data _outside of_ topology, I think it's fine. Properly mocking it would probably require setting up some yaml files for a dummy VO and resource, which would take too long.